README.md

```diff
@@ -42,6 +42,7 @@ Help welcome, eg by filling in the entries with `?` `TODO` and `CHECKME`, correc
 | User defined operators | partial: opCall opSlice, opAssign etc | yes | -1 |
 | User defined attributes | yes | ? | ? |
 | RAII | yes (modulo caveats https://github.com/timotheecour/D_vs_nim/issues/27) | no, see: [RAII](https://forum.nim-lang.org/t/362/1) | 1 |
+| prevention of null dereferencing | <p>manual null checks required</p> <p>`null` pointers to class objects can exist. The only way to check for them is to remember to use `if (myObj is null)`. Lessening the impact of this, many built-in types, such as integers, arrays, maps, and struct objects, can never be `null` ([source](https://w0rp.com/blog/post/null-and-d-programming-language/)).</p> | <p>manual null checks required</p> <p>`nil` references and pointers can exist. The standard way to check for them is to remember to use `if myObj.isNil`. Lessening the impact of this, many built-in types can never be `nil`, including `seq`s. However, tables (which are like D’s maps) can be `nil`.</p> <p>Nim also has an experimental `notnil` feature that enables a [`not nil` type annotation](https://nim-lang.org/docs/manual.html#types-not-nil-annotation), e.g. `ref SomeObject not nil`, to ensure that a variable can never hold `nil`. However, it doesn’t seem to be well-supported. The `notnil` feature was made experimental [in 2018](https://github.com/nim-lang/Nim/blob/devel/changelogs/changelog_0_19_0.md#changes-affecting-backwards-compatibility), but in 2020 its [brief documentation](https://nim-lang.org/docs/manual.html#types-not-nil-annotation) is still in the wrong section of the manual, and it has at least one [open bug](https://github.com/nim-lang/Nim/issues/7248) that may let a `nil` dereference error happen with a `not nil` type. `not nil` types are also [not widely used](https://github.com/nim-lang/Nim/pull/3296#issuecomment-138483068) in the standard library even for procs that will never return `nil`, so users must make redundant assertions.</p> | ? |
 | **debugging** |
 | **maturity** |
```

Comment:
> However, it doesn’t seem to be well-supported. The notnil feature was made experimental in 2018, but in 2020 its brief documentation is still in the wrong section of the manual, and it has at least one open bug that may let a nil dereference error happen with a not nil type. not nil types are also not widely used in the standard library even for procs that will never return nil, so users must make redundant assertions.


AFAIK https://github.com/nim-lang/Nim/issues/7248 isn't related to `not nil` type annotation; the code snippet doesn't have `not nil` annotation.
I would just skip this whole sentence, it's enough to say that `not nil` annotation is still an experimental feature.
That being said, PR is welcome (in nim) to move not nil discussion to experimental section of manual.

Reply:
Shortened to two sentences:

> Nim also has an experimental `notnil` feature that enables a [`not nil` type annotation](https://nim-lang.org/docs/manual.html#types-not-nil-annotation), e.g. `ref SomeObject not nil`, to ensure that a variable can never hold `nil`. This feature was made experimental [in 2018](https://github.com/nim-lang/Nim/blob/devel/changelogs/changelog_0_19_0.md#changes-affecting-backwards-compatibility).

I didn’t make the PR for the Nim manual yet, so this link still points to the old section of the manual.